### PR TITLE
Add Github CHERRY_PICK pull request template

### PR DESCRIPTION
**Pull request by Kyle-Ye**
Bug/issue #, if applicable: 

## Summary

- Add Github CHERRY_PICK pull request template
- Update gitignore file

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE/CHERRY_PICK.md
Comment (Kyle-Ye) [reply .github/PULL_REQUEST_TEMPLATE/CHERRY_PICK.md:1]: I did not notice that. Thank you for reminding me of it. I'll update to follow it.

**Comment by Kyle-Ye** [comment]
Combined the template from https://forums.swift.org/t/swift-5-7-release-process/56316#pull-requests-for-release-branch-7 and https://github.com/apple/swift-docc/pull/248.

File: .github/PULL_REQUEST_TEMPLATE/CHERRY_PICK.md
Comment (franklinsch) [reply .github/PULL_REQUEST_TEMPLATE/CHERRY_PICK.md]: Sorry for the nitpick here, but I think we should keep the exact same formatting outline in the forum post, so that it's consistent with other Swift repos:

- **Explanation**: _A description of the issue being fixed or enhancement being made. This can be brief, but it should be clear._
- **Scope**: _An assessment of the impact/importance of the change. For example, is the change a source-breaking language change, etc._
- **GitHub Issue**: _The issue number if the change fixes/implements an issue/enhancement on [GitHub Issues](https://github.com/apple/swift-docc/issues)._
- **Risk**: _What is the (specific) risk to the release for taking this change?_
- **Testing**: _What specific testing has been done or needs to be done to further validate any impact of this change?_
- **Reviewer**: _One or more code owners for the impacted components should review the change. Technical review can be delegated by a code owner or otherwise requested as deemed appropriate or useful._

Comment (Kyle-Ye) [reply .github/PULL_REQUEST_TEMPLATE/CHERRY_PICK.md]: Got it. 👌

**Comment by Kyle-Ye** [comment]
@swift-ci please test

File: .github/PULL_REQUEST_TEMPLATE/CHERRY_PICK.md
Comment (franklinsch) [inline .github/PULL_REQUEST_TEMPLATE/CHERRY_PICK.md:11]:
```suggestion
- **Explanation**: _A description of the issue being fixed or enhancement being made. This can be brief, but it should be clear._
- **Scope**: _An assessment of the impact/importance of the change. For example, is the change a source-breaking language change, etc._
- **GitHub Issue**: _The issue number if the change fixes/implements an issue/enhancement on [GitHub Issues](https://github.com/apple/swift-docc/issues)._
- **Risk**: _What is the (specific) risk to the release for taking this change?_
- **Testing**: _What specific testing has been done or needs to be done to further validate any impact of this change?_
- **Reviewer**: _One or more code owners for the impacted components should review the change. Technical review can be delegated by a code owner or otherwise requested as deemed appropriate or useful._
```

I think the way GitHub Markdown parsing works, it'll add unnecessary changes newlines between the list items. Otherwise this looks great!

**Review by franklinsch — APPROVED** [review]
Thank you @Kyle-Ye!

**Comment by Kyle-Ye** [comment]
@swift-ci please test and merge

**Comment by Kyle-Ye** [comment]
@swift-ci please test

**Comment by Kyle-Ye** [comment]
Unfortunately there is not a GUI list like Github issue for us to choose when create a Pull Request on Github website.

**Comment by franklinsch** [comment]
Ooh I see, that's too bad. I think in that case I prefer the previous behaviour where you get just the default template; the point was that you have something to fill in if you're filing a new PR. What do you think? Should we revert the change?

**Comment by franklinsch** [comment]
Ok, if that'll take some time, should we revert the change for now and if the config file works, re-introduce the templates with the extra file?

**Comment by Kyle-Ye** [comment]
There are many people encountering the same problem. Here is a solution

https://github.community/t/is-there-a-pull-request-template-selector-similar-to-issues/171838/9

If you accept, we can move DEFAULT.md to its original location and name. Then the default behavior will come back

**Comment by Kyle-Ye** [comment]
> Looking into this a little more, I realized there’s a way to have a default template, as well as additional templates. The default template is stored directly into the .github folder and called PULL_REQUEST_TEMPLATE.MD. We have an additional PULL_REQUEST_TEMPLATES folder containing additional templates that can be used.

> Back in 2014 / 2015, using the template query-param was the original way to access templated issues. Since then we got the issues/new/choose endpoint for issue templates and then we got [yml forms 3](https://gh-community.github.io/issue-template-feedback/welcome/) but pull request templates are still using the old, old way.

**Comment by franklinsch** [comment]
I think let's revert to the original single template, but add a commented out section at the bottom with the template with the cherry-pick instructions:

```md
## Checklist

…

<!--

Note: If this PR cherry-picks a change for a release branch, use the following template instead:

- **Explanation**: _A description of the issue being fixed or enhancement being made. This can be brief, but it should be clear._
- **Scope**: _An assessment of the impact/importance of the change. For example, is the change a source-breaking language change, etc._
- **GitHub Issue**: _The issue number if the change fixes/implements an issue/enhancement on [GitHub Issues](https://github.com/apple/swift-docc/issues)._
- **Risk**: _What is the (specific) risk to the release for taking this change?_
- **Testing**: _What specific testing has been done or needs to be done to further validate any impact of this change?_
- **Reviewer**: _One or more code owners for the impacted components should review the change. Technical review can be delegated by a code owner or otherwise requested as deemed appropriate or useful._
-->

```

What do you think? That seems to be the cleanest solution.

**Comment by Kyle-Ye** [comment]
Revert the PR now is totally OK. But I disagree with the approach you describe.

Github has multi PR template feature supported. If we have a need for it, we should follow the Github's official spec instead of using textual form to workaround which seems error-prone for me.

1. On `ph` Github CLI, we can choose different PR template in terminal
2. On Github's website, we can only use the oldest query-param way to choose it.

> Back in 2014 / 2015, using the template query-param was the original way to access templated issues. Since then we got the issues/new/choose endpoint for issue templates and then we got [yml forms 3](https://gh-community.github.io/issue-template-feedback/welcome/) but pull request templates are still using the old, old way.

If we want to revert to the original behevior, we can move the DEFAULT.md to its original location. Then when creating a PR on Github's website, it will use it as the default template.

And also I think we should still put the extra template on PULL_REQUEST_TEMPLATES folder for the following reasons:
1. We have `gh` CLI to utilize it
2. Eventually I believe Github website will support it like issues. And since we are following the spec, no change is needed. We get it free.

**Comment by franklinsch** [comment]
We still lose discoverability of the cherry-pick template though, at least on the web, and for now. Maybe we can do something in the middle as well, where the comment I'm proposing to add at the bottom can instead refer to the other template:

```
<!--

If you're opening a PR to cherry-pick a change for a release branch, use this template instead:
https://github.com/apple/swift-docc/blob/main/.github/PULL_REQUEST_TEMPLATE/CHERRY_PICK.md

-->
```

**Comment by Kyle-Ye** [comment]
> Back in 2014 / 2015, using the template query-param was the original way to access templated issues. Since then we got the issues/new/choose endpoint for issue templates and then we got [yml forms 3](https://gh-community.github.io/issue-template-feedback/welcome/) but pull request templates are still using the old, old way.

By the way, there is a PR #232 to update the issue template to the latest yaml form so that we can fill in issues more easily(In a GUI friendly way).

> Context:
> v1: accept query params for specfying template
> v2: a GUI list chooser to choose different template
> v3: GUI composer(md template-> yml template)
> Issue is now at v3, but Pull Request is still at v1

**Pull request merged**
